wolfcrypt/test/test.c

```diff
@@ -39968,14 +39968,26 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t cmac_test(void)
 
         XMEMSET(tag, 0, sizeof(tag));
         tagSz = sizeof(tag);
+#if !defined(HAVE_FIPS) || \
```

Comment:
Hrmmm... Not sure why this test is limited by FIPS.  Shouldn't we be performing the _ex test when cryptocb is defined rather than FIPS?  I haven't reviewed the other chunks of this test code. Currently, FIPS is not compatible with cryptocb so we may need to add a second test that explicitly works the cryptocb version (devid!=INVALID_DEVID) of this code .

Reply:
So I just followed the existing pattern of how `wc_InitCmac` vs `wc_initCmac_ex` was called earlier in this test. And to be clear, the `_ex` is called only when FIPS is NOT on, my guess being because FIPS isn't compat with cryptoCb. But I agree, you would think it would be gated based on whether cryptoCb is on, not FIPS, leaving the incompatibility of the two as a compile time detection error.

Elsewhere in `test.c` the FIPS check isn't even present and it just calls the `_ex` versions of crypto functions regardless, relying on the fact that internal cryptoCb invocations will be gated out and `_ex` will behave like normal if cryptoCb support isn't turned on. 

We should probably just remove the FIPS check and call the _ex function regardless, as cryptoCb code will not be compiled if support for it isn't turned on.

> we may need to add a second test that explicitly works the cryptocb version (devid!=INVALID_DEVID) of this code.

This should already be happening. When cryptoCb support is turned on, the `cryptocb_test()` function invokes this function (`cmac_test()`), but with a valid cryptoCb registered. I verified the cryptoCb code paths were being hit.

Reply:
Figured it out - the call is gated by that particular version of FIPS because at that point in time, the API did not exist. So when our Jenkins FIPS tests run (and replace cmac.{c,h} with their FIPS implementations from the FIPS repo) then this wolfCrypt test will be missing that symbol.